File: sites/python/server/mailer.js
'use strict';

/**
 * Email notifications for new leads.
 * Requires nodemailer + Gmail App Password in .env.
 *
 * Gmail setup:
 *   1. Enable 2-Step Verification on Google account
 *   2. Go to myaccount.google.com → Security → App passwords
 *   3. Create App Password → copy 16-char code → paste as SMTP_PASS in .env
 */

let transporter = null;

function getTransporter() {
  if (transporter) return transporter;

  const nodemailer = require('nodemailer');

  if (!process.env.SMTP_USER || !process.env.SMTP_PASS) {
    return null;
  }

  transporter = nodemailer.createTransport({
    host:   process.env.SMTP_HOST || 'smtp.gmail.com',
    port:   parseInt(process.env.SMTP_PORT || '587'),
    secure: false,
    auth: {
      user: process.env.SMTP_USER,
      pass: process.env.SMTP_PASS,
    },
  });

  return transporter;
}

const COURSE_NAMES = {
  scratch: 'Scratch (6–10 років)',
  python:  'Python (10–14 років)',
  roblox:  'Roblox (9–14 років)',
  web:     'Веб-розробка (12–16 років)',
  help:    'Допоможіть обрати',
};

/**
 * Send notification email for a new lead.
 * Fails silently — lead is already saved to DB regardless.
 */
async function sendLeadNotification(lead) {
  const t = getTransporter();
  if (!t) return;

  const to   = process.env.NOTIFY_EMAIL || process.env.SMTP_USER;
  const from = `"My Computer Academy" <${process.env.SMTP_USER}>`;

  const courseName = COURSE_NAMES[lead.course] || lead.course || '—';
  const time       = new Date().toLocaleString('uk-UA', { timeZone: 'Europe/Kiev' });

  const html = `
<!DOCTYPE html>
<html lang="uk">
<head><meta charset="UTF-8"/></head>
<body style="font-family:Arial,sans-serif;background:#f5f5f5;padding:20px">
  <div style="max-width:520px;margin:0 auto;background:#fff;border-radius:10px;overflow:hidden;box-shadow:0 2px 8px rgba(0,0,0,.1)">
    <div style="background:#6C47FF;padding:20px 28px">
      <h2 style="color:#fff;margin:0;font-size:18px">🎓 Нова заявка — My Computer Academy</h2>
    </div>
    <div style="padding:24px 28px">
      <table style="width:100%;border-collapse:collapse">
        <tr><td style="padding:8px 0;color:#888;width:140px">Ім'я дитини</td><td style="padding:8px 0;font-weight:600">${lead.child_name}</td></tr>
        <tr><td style="padding:8px 0;color:#888">Вік</td><td style="padding:8px 0">${lead.age || '—'} років</td></tr>
        <tr><td style="padding:8px 0;color:#888">Курс</td><td style="padding:8px 0">${courseName}</td></tr>
        <tr><td style="padding:8px 0;color:#888">Телефон</td><td style="padding:8px 0"><strong><a href="tel:${lead.phone}" style="color:#6C47FF;text-decoration:none">${lead.phone}</a></strong></td></tr>
        <tr><td style="padding:8px 0;color:#888">Email</td><td style="padding:8px 0">${lead.email ? `<a href="mailto:${lead.email}" style="color:#6C47FF">${lead.email}</a>` : '—'}</td></tr>
        <tr><td style="padding:8px 0;color:#888">Заявка №</td><td style="padding:8px 0">#${lead.id}</td></tr>
        <tr><td style="padding:8px 0;color:#888">Час</td><td style="padding:8px 0">${time}</td></tr>
      </table>
    </div>
    <div style="background:#f9f7ff;padding:16px 28px;border-top:1px solid #eee">
      <p style="margin:0;color:#888;font-size:13px">Передзвоніть протягом 30 хвилин 📞</p>
    </div>
  </div>
</body>
</html>`;

  const text = `Нова заявка #${lead.id}
Дитина: ${lead.child_name}, ${lead.age || '—'} років
Курс: ${courseName}
Телефон: ${lead.phone}
Email: ${lead.email || '—'}
Час: ${time}`;

  try {
    await t.sendMail({
      from,
      to,
      subject: `📩 Нова заявка: ${lead.child_name} — ${courseName}`,
      text,
      html,
    });
    console.log(`[MAIL] Notification sent to ${to} for lead #${lead.id}`);
  } catch (err) {
    console.error(`[MAIL ERROR] Failed to send notification:`, err.message);
  }
}

module.exports = { sendLeadNotification };

File: sites/old/server/orders/mailer.py
import os
import smtplib
from email.header import Header
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText
from datetime import datetime
from html import escape


def send_lead_notification(order):
    """
    Send an HTML email notification for a new lead, mirroring the Node
    landings' mailer.js pattern (same Gmail account, same layout).
    Fails silently — the lead is already saved locally regardless.
    """
    smtp_user = os.environ.get('SMTP_USER', '')
    smtp_pass = os.environ.get('SMTP_PASS', '')
    if not smtp_user or not smtp_pass:
        print('Email notification not configured: SMTP_USER or SMTP_PASS missing')
        return

    smtp_host = os.environ.get('SMTP_HOST', 'smtp.gmail.com')
    smtp_port = int(os.environ.get('SMTP_PORT', '587'))
    to_addr = os.environ.get('NOTIFY_EMAIL', smtp_user)

    time_str = datetime.now().strftime('%d.%m.%Y %H:%M')

    # Escape before interpolating into HTML — order.name/message are raw
    # user input and were previously inserted unescaped, so a lead message
    # containing HTML/script tags rendered live in whoever's mail client
    # opened the notification.
    safe_name = escape(order.name)
    safe_phone = escape(order.phone)
    safe_message = escape(order.message)

    html = f"""\
<!DOCTYPE html>
<html lang="uk">
<head><meta charset="UTF-8"/></head>
<body style="font-family:Arial,sans-serif;background:#f5f5f5;padding:20px">
  <div style="max-width:520px;margin:0 auto;background:#fff;border-radius:10px;overflow:hidden;box-shadow:0 2px 8px rgba(0,0,0,.1)">
    <div style="background:#242538;padding:20px 28px">
      <h2 style="color:#fff;margin:0;font-size:18px">Нова заявка — old.mycomputer.education</h2>
    </div>
    <div style="padding:24px 28px">
      <table style="width:100%;border-collapse:collapse">
        <tr><td style="padding:8px 0;color:#888;width:140px">Ім'я</td><td style="padding:8px 0;font-weight:600">{safe_name}</td></tr>
        <tr><td style="padding:8px 0;color:#888">Телефон</td><td style="padding:8px 0"><strong><a href="tel:{safe_phone}" style="color:#242538;text-decoration:none">{safe_phone}</a></strong></td></tr>
        <tr><td style="padding:8px 0;color:#888">Повідомлення</td><td style="padding:8px 0">{safe_message}</td></tr>
        <tr><td style="padding:8px 0;color:#888">Заявка №</td><td style="padding:8px 0">#{order.id}</td></tr>
        <tr><td style="padding:8px 0;color:#888">Час</td><td style="padding:8px 0">{time_str}</td></tr>
      </table>
    </div>
    <div style="background:#f5f5f7;padding:16px 28px;border-top:1px solid #eee">
      <p style="margin:0;color:#888;font-size:13px">Передзвоніть протягом 30 хвилин</p>
    </div>
  </div>
</body>
</html>"""

    text = (
        f"Нова заявка #{order.id}\n"
        f"Ім'я: {order.name}\n"
        f"Телефон: {order.phone}\n"
        f"Повідомлення: {order.message}\n"
        f"Час: {time_str}"
    )

    msg = MIMEMultipart('alternative')
    msg['From'] = f'My Computer Academy <{smtp_user}>'
    msg['To'] = to_addr
    # Header() strips/encodes anything that could inject extra headers via
    # CR/LF in a lead's name — raw f-string interpolation didn't.
    msg['Subject'] = Header(f'Нова заявка (old): {order.name}'.replace('\r', ' ').replace('\n', ' '), 'utf-8')
    msg.attach(MIMEText(text, 'plain', 'utf-8'))
    msg.attach(MIMEText(html, 'html', 'utf-8'))

    try:
        with smtplib.SMTP(smtp_host, smtp_port, timeout=10) as server:
            server.starttls()
            server.login(smtp_user, smtp_pass)
            server.sendmail(smtp_user, [to_addr], msg.as_string())
        print(f'[MAIL] Notification sent to {to_addr} for order #{order.id}')
    except Exception as e:
        print(f'[MAIL ERROR] Failed to send notification: {e}')
